include/ck/tensor_operation/gpu/thread/threadwise_tensor_slice_transfer_v3r1.hpp

```diff
@@ -337,8 +338,27 @@ struct ThreadwiseTensorSliceTransfer_v3r1
 
                 // do data transpose
                 // TODO type_convert is not used yet!!!!!
-                transpose_vectors<SrcData, DstScalarPerVector, SrcScalarPerVector>{}(
-                    src_vector_refs, dst_vector_refs);
+                transpose_convert_vectors<SrcData,
+                                          DstData,
+                                          DstScalarPerVector,
+                                          SrcScalarPerVector>{}(src_vector_refs, dst_vector_refs);
+            });
+        }
+        else if constexpr(SrcVectorDim == DstVectorDim && SrcScalarPerVector % 2 == 0 &&
+                          DstScalarPerVector % 2 == 0 &&
+                          is_same<half_t, remove_cvref_t<SrcData>>::value &&
+                          is_same<bhalf_t, remove_cvref_t<DstData>>::value)
+        {
+            auto NewSliceLengths = SliceLengths{}.template Modify(
```

Comment:
Have we considered doing conversion after the optional transpose? This should probably eliminate another bf16 variant of `transpose_convert_vectors` over the existing fp16 variant

Reply:
yes, it is a good idea, doing that can make convert code and transpose  one function each.
but it maybe need to change a lot of existing code and not sure fast as current code, so I don't want to change it

Reply:
I'm not 100% sure, but if we don't do transpose and convert at the same time, we will need more instructions. Is that correct, @ltqin, @rosenrodt  ?

Reply:
The current implementation is the leanest it seems:

### 2x2 transpose & convert at the same time
v_cvt_f32_f16
v_cvt_f32_f16
v_pack // pack high bits to get bf16x2
v_cvt_f32_f16
v_cvt_f32_f16
v_pack // pack high bits to get bf16x2

### 2x2 transpose then convert
v_pack
v_pack 

v_cvt_f32_f16
v_cvt_f32_f16
v_pack // pack high bits to get bf16x2
v_cvt_f32_f16
v_cvt_f32_f16
v_pack // pack high bits to get bf16x2

So there's 2 extra VALUs for each 2x2 fp16 tile

Reply:
I think the conversation is resolved then

Reply:
@rosenrodt Please mark the conversation has "resolved" if you think there is no more issue